samples/drivers/spi_nand/src/main.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2022 Macronix International Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <drivers/flash.h>
+#include <device.h>
+#include <devicetree.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if DT_NODE_HAS_STATUS(DT_INST(0, st_stm32_qspi_nand), okay)
+#define FLASH_DEVICE DT_LABEL(DT_INST(0, st_stm32_qspi_nand))
+#define FLASH_NAME "ST STM32_QSPI-NAND"
+#elif DT_NODE_HAS_STATUS(DT_INST(0, jedec_spi_nand), okay)
+#define FLASH_DEVICE DT_LABEL(DT_INST(0, jedec_spi_nand))
+#define FLASH_NAME "JEDEC SPI-NAND"
+#else
+#error Unsupported flash driver
+#endif
+
+#if defined(CONFIG_BOARD_ADAFRUIT_FEATHER_STM32F405)
+#define FLASH_TEST_REGION_OFFSET 0xf000
+#elif defined(CONFIG_BOARD_ARTY_A7_ARM_DESIGNSTART_M1) || \
+	defined(CONFIG_BOARD_ARTY_A7_ARM_DESIGNSTART_M3)
+/* The FPGA bitstream is stored in the lower 536 sectors of the flash. */
+#define FLASH_TEST_REGION_OFFSET \
+	DT_REG_SIZE(DT_NODE_BY_FIXED_PARTITION_LABEL(fpga_bitstream))
+#else
+#define FLASH_TEST_REGION_OFFSET 0xff000
+#endif
```

Comment:
To be kept ?

Reply:
I have fixed the above issues.
Thanks for your review.